torchaudio/backend/sox_io_backend.py

```diff
@@ -77,4 +72,58 @@ def load(
     return signal.get_tensor(), signal.get_sample_rate()
 
 
+@_mod_utils.requires_module('torchaudio._torchaudio')
+def save(
+        filepath: str,
+        tensor: torch.Tensor,
+        sample_rate: int,
+        channels_first: bool = True,
+        compression: Optional[float] = None,
+        frames_per_chunk: int = 65536,
+):
+    """Save audio data to file.
+
+    Supported formats are;
+     - WAV
+        - 32-bit floating-point
+        - 32-bit signed integer
+        - 16-bit signed integer
+        -  8-bit unsigned integer
+     - MP3
+     - FLAC
+     - OGG/VORBIS
+
+    Args:
+        filepath: Path to save file.
+        tensor: Audio data to save. must be 2D tensor.
+        sample_rate: sampling rate
+        channels_first: If True, the given tensor is interpreted as ``[channel, time]``.
+        compression: Used for formats other than WAV. This corresponds to ``-C`` option
+            of ``sox`` command.
+            See the detail at http://sox.sourceforge.net/soxformat.html.
+            - MP3: Either bitrate [kbps] with quality factor, such as ``128.2`` or
+                VBR encoding with quality factor such as ``-4.2``. Default: ``-4.5``
+            - FLAC: compression level. Whole number from ``0`` to ``8``.
+                ``8`` is default and highest compression.
+            - OGG/VORBIS: number from -1 to 10; -1 is the highest compression and lowest
+                quality. Default: ``3``.
+        frames_per_chunk: The number of frames to process (convert to ``int32`` internally
+            then write to file) at a time.
+    """
+    if compression is None:
+        ext = str(filepath)[-3:].lower()
+        if ext == 'wav':
+            compression = 0.
+        elif ext == 'mp3':
+            compression = -4.5
+        elif ext == 'flac':
+            compression = 8.
+        elif ext in ['ogg', 'vorbis']:
+            compression = 3.
+        else:
+            raise RuntimeError(f'Unsupported file type: "{ext}"')
```

Comment:
@vincentqb I thought of calling for opening an issue for adding support for other audio formats, but differentiating the valid extensions that are supported by libsox from invalid ones is not trivial and there are many formats that require additional libraries to be installed. Meaning that adding support on our `get_signalinfo` function may not be enough for users to actually use a new format, depending on the format. So I think we want to be more cautious and not giving high expectations to users casually. Therefore I decided not to include call for opening a PR in the error message.

Reply:
Can you remind me how the current backend exposes more backend? From our discussion, the user had to provide extra internal information, right?

Reply:
The existing implementation asks users to construct the following data structure, which requires users to know the internal of libsox.

https://fossies.org/dox/sox-14.4.2/structsox__signalinfo__t.html
https://fossies.org/dox/sox-14.4.2/structsox__encodinginfo__t.html